node_list.go

```diff
@@ -1,46 +1,62 @@
 package memcacheha
 
+import "sync"
+
 // NodeList represents a list of memcache servers configured/discovered by this client.
 type NodeList struct {
 	Nodes map[string]*Node
```

Comment:
Is it reasonable to make `Nodes` private?  This would ensure you know everywhere the map is being accessed - which would in turn enable you to use the mutex to protect all access.

Reply:
Yes that would be much better, but also requires a fairly large refactor that I'm not sure we should do now...definitely at some point in the future though!